model-engine/model_engine_server/inference/async_inference/tasks.py

```diff
@@ -46,6 +49,11 @@ def init_worker_global():
         default_callback_url=endpoint_config.default_callback_url,
         default_callback_auth=endpoint_config.default_callback_auth,
         monitoring_metrics_gateway=DatadogInferenceMonitoringMetricsGateway(),
+        endpoint_id=endpoint_config.endpoint_id,
+        endpoint_type=endpoint_config.endpoint_type,
+        bundle_id=endpoint_config.bundle_id,
+        labels=endpoint_config.labels,
+        streaming_storage_gateway=FirehoseStreamingStorageGateway(),
```

Comment:
yeah that could work

Reply:
seems like this doesn't work, I can't import from `api` in `inference`

Reply:
wait is this where the async forwarder is implemented? if not, I think this code might be dead code (i.e. it was used back when async task receive/push-result would run in the same container as the user code

Reply:
if the code is dead code, then I think the import would work because there was at some point some amount of copying only part of the code to the user inference container because of security reasons

Reply:
yeah this is dead code so the import works but I'm referring to everywhere else (like in forwarding.py) it wouldn't work. also, if this is dead code anyways, then there's not really a point in importing from external interfaces?

Reply:
IMO we could probably just delete the dead code (as long as we're sure it's dead)

re forwarding.py I think it should be possible to make ExternalInterfaces importable; I guess since we have stuff in `inference/{domain|infra}/gateways` the current layout of things is a bit different; I still think that in the ideal state we'd be able to use the same plugins mechanism to control which interfaces we're using

Reply:
hm how do we make it possible to be importable in forwarding.py? yeah ideally it does make sense to control which interfaces we're using.

Reply:
I think for now I'll leave the 2 gateways like this and can always follow-up with deleting dead code / using ExternalInterfaces